File: dalat-be/src/main/java/com/dan/dalat/models/UserFeedback.java
package com.dan.dalat.models;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDateTime;

@Entity
@Table(name = "user_feedback")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class UserFeedback {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    User user;

    @ManyToOne
    @JoinColumn(name = "place_id")
    Place place;

    int rating; // 1-5 sao
    @Column(columnDefinition = "TEXT")
    String comment;

    @Column(nullable = false, updatable = false)
    @Temporal(TemporalType.TIMESTAMP)
    LocalDateTime createdAt;
}

File: dalat-filter/models/user_feedback.py
from database import Base
from sqlalchemy import Column, Integer, String, Float, Text

class UserFeedback(Base):
    __tablename__ = "user_feedback"

    id = Column(Integer, primary_key=True, index=True)
    comment = Column(String, nullable=False)
    rating = Column(Integer, nullable=False)
    place_id = Column(Integer, nullable=False)
    user_id = Column(Integer, nullable=False)
    created_at = Column(String, nullable=False)